setup.py

```diff
@@ -11,12 +11,11 @@
 from setuptools import setup, find_packages
 
 
-project_path = lambda *names: os.path.join(os.path.dirname(__file__), *names)
+longdesc = "\n\n".join((open("README.rst").read(),
+                        open("ABOUT.rst").read()))
 
-longdesc = "\n\n".join((open(project_path("README.txt")).read(),
-                        open(project_path("ABOUT.txt")).read()))
-
-root_files = glob.glob(project_path("*.txt"))
+root_files = glob.glob("*.txt")
+root_files.extend(glob.glob("*.rst"))
 data_files = [("", [name for name in root_files
```

Comment:
Das Ziel Deiner Änderungen bzgl. `project_path` war ja, absolute Pfade aus den `root_files` wegzubekommen. Ich würde es besser finden, alle Zugriffe auf Dateien zum Auslesen und alles Globbing wie bisher mit absoluten Pfaden durchzuführen und die Verzeichnisse erst beim Zusammensetzen der `data_files` abzuschneiden. Vielleicht weißt Du ja von Garantien für die Ausführung von `setup.py`, aber ohne jetzt viel ausprobiert zu haben, würde ich erst mal nicht ausschließen wollen, daß das Arbeitsverzeichnis je nach Installationswerkzeug auch mal ein anderes als das Projektverzeichnis sein kann. Wenn das der Fall ist, dann produziert `setup.py` jedenfalls Fehler, sobald es z.B. kein `README.rst` finden kann.

Reply:
Ich habe vielleicht 10 gocept-Pakete angepasst und dort überall die Verwendung von absoluten Pfaden entfernt, danach hat es in keinem Fall Probleme gegeben. In der Regel wird ein Paket ja entweder mit `pip` oder `buildout` installiert und die kommen mit relativen Pfaden klar.

Wenn man in einem `virtualenv` versucht, `tl.eggdeps` zu installieren, wird – neben der Fehlermeldung – folgender Text angezeigt:

>   setup() arguments must *always* be /-separated paths relative to the
>  setup.py directory, *never* absolute paths.

Das war für mich der Grund, alle absoluten Pfade rauszureißen.

Reply:
@tlotze Auch bei mir trat der Fehler auf, die Installation durch buildout war ohne macs Branch unmöglich.